### PR TITLE
Restrict ConfigMap to single namespace plus security improvements

### PR DESCRIPTION
**Pull request by pjbgf**
- Add RBAC for service principal
- Add usage and fix example
- Drop caps and no_new_privs
- Restrict configmaps to a single namespace

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -27,13 +27,15 @@ import (
 )
 
 var (
-	sync     = time.Second * 30
-	setupLog = ctrl.Log.WithName("setup")
+	sync                    = time.Second * 30
+	setupLog                = ctrl.Log.WithName("setup")
+	namespaceToWatch string = "security-operators"
```
Comment (hasheddan) [inline main.go:32]: Do we want this to be configurable?

File: deploy/operator.yaml [hunk 2]
```diff
@@ -29,10 +53,14 @@ spec:
           image: "securityoperators/seccomp-operator:latest"
           imagePullPolicy: Always
           volumeMounts:
-          - mountPath: /var/lib/kubelet/seccomp
-            name: seccomp-root-path
+          - mountPath: /var/lib/kubelet
+            name: kubelet-root-path
+          securityContext:
+            allowPrivilegeEscalation: false
```
Comment (saschagrunert) [inline deploy/operator.yaml:59]: That’s the default right? If so I guess we can remove it.

Comment (pjbgf) [reply deploy/operator.yaml:59]: It actually defaults to true: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privilege-escalation

File: main.go (hunk 1 above)
Comment (pjbgf) [reply main.go:32]: Is it worth also making reconciliation intervals configurable?

File: deploy/operator.yaml (hunk 2 above)
Comment (saschagrunert) [reply deploy/operator.yaml:59]: Whoot.

File: main.go (hunk 1 above)
Comment (pjbgf) [reply main.go]: Fixed it - although I am not sure about the ENV name I used: `NAMESPACE_TO_WATCH`.

**Comment by saschagrunert** [comment]
Thank you!

**Pull request merged**